src/shape.py

```diff
@@ -76,3 +76,65 @@ def create_shape(type: str, start_row: int, start_col: int) -> Shape:
         return SquareShape(start_row, start_col)
 
     raise NotImplementedError("Shape type not implemented")
+
+
+"""
+    U - - 
+    U - - 
+    U U -
+
+    U U U
+    U - -
+    - - -
+
+    - U U
+    - - U
+    - - U
+
+    - - -
+    - - U
+    U U U
+    """
+
+
+def rotate(shape: Shape) -> Shape:
+    unit_row_array = []
+    unit_col_array = []
+
+    for unit in shape.units:
+        unit_row_array.append(unit.row)
+        unit_col_array.append(unit.col)
+
+    row_diff = max(unit_row_array) - min(unit_row_array) + 1
+    col_diff = max(unit_col_array) - min(unit_col_array) + 1
+    min_value = min(min(unit_row_array), min(unit_col_array))
+    dimension = max(row_diff, col_diff)
```

Comment:
max ve mini surekli ic ice yazmaktansa birer degiskene atayipda yapabiliriz asagiya bir tane ornek birakiyorum

```suggestion
    unit_rows = [unit.row for unit in shape.units]
    unit_cols = [unit.col for unit in shape.units]
    
    max_row, min_row = max(unit_rows), min(unit_rows)
    max_col, min_col = max(unit_cols), min(unit_cols)

    row_diff = max_row - min_row + 1
    col_diff = max_col - min_col + 1
    min_value = min(min_col, min_row)
    dimension = max(row_diff, col_diff)
```